core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -300,6 +304,7 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
       // Ensures writes to the data container use the right consistent hash
       // No need for a try/finally block, since it's just an assignment
       stateTransferLock.acquireExclusiveTopologyLock();
+      triangleOrderManager.updateCacheTopology(cacheTopology);
```

Comment:
in more detail, the ```isNext``` is only invoked when the topology id is higher or the same as the command topology id. This is ensured by the ReadAction mechanism that waits for the topology before performing the next checks.